### PR TITLE
Azure Spatial Anchors Spatial Coordinate Implementation (UWP, Android)

### PR DESCRIPTION
**Pull request by andreiborodin**
This change implements the ISpatialCoordinateService for the Azure Spatial Anchors. This change has a few caveats:

- Due to structure of MRTK project, ASA must be imported to make it work
- Furthermore, ASA is imported into the root of the project, meaning this code can't be in any of the ASMDEFs as ASMDEFs compile first, and the reference is not available

There are a few additional things that will need to be done before change goes in.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerSpatialCoordinateService.cs [hunk 1]
```diff
@@ -765,34 +765,35 @@ protected void EstablishLocalOrigin()
...
-                        _aRPointCloudManager.pointCloudUpdated -= OnPointCloudUpdated;
-                        _listeningToPointCloudChanges = false;
-                    }
-                }
-                else
-                {
-                    Debug.Log("Point cloud did not contain enough points to establish a local origin, current size: " + points.Count);
-                }
-            }
+            throw new NotImplementedException();
```
Comment (chrisfromwork) [inline Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerSpatialCoordinateService.cs]: we could probably require a specific mobile spectator preprocessor directive compared to commenting this out.

Comment (andreiborodin) [reply Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerSpatialCoordinateService.cs]: Yep, will do that before I check-in, let me know which pre-processor.

**Comment by chrisfromwork** [comment]
I'm going to pick up these commits and merge this payload into my work monday. Starting next week my first priority will be getting this up and running with code organization that can remain usable by dslr work.

File: Assets/MixedRealityToolkit.Extensions/SpectatorView/Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.asmdef [hunk 2]
```diff
@@ -1,21 +1,20 @@
 {
-  "name": "Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView",
-  "references": [
-    "Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection",
-    "Microsoft.MixedReality.Toolkit.Extensions.PhotoCapture",
-    "Microsoft.MixedReality.Toolkit.Extensions.Experimental.ScreenRecording",
-    "Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing",
-    "Unity.XR.ARFoundation",
```
Comment (andreiborodin) [inline Assets/MixedRealityToolkit.Extensions/SpectatorView/Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.asmdef]: Including this by default gave me some issues with GoogleARCore, we should have this in instructions and tell user to then go set "AR_FOUNDATION" scripting define symbol.

File: Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerSpatialCoordinateService.cs (hunk 1 above)
Comment (andreiborodin) [reply Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerSpatialCoordinateService.cs]: Added pre-processor "AR_FOUNDATION"

**Comment by andreiborodin** [comment]
Awesome. And yeah, this last iteration fixes some bugs in localization :)

---
In reply to: [495729318](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4544#issuecomment-495729318) [](ancestors = 495729318)

**Pull request closed without merging**